cli/src/merge_tools/builtin.rs

```diff
@@ -521,7 +526,23 @@ fn apply_changes(
                 tree_builder.set_or_remove(path, value);
             }
             scm_record::SelectedContents::Text { contents } => {
-                let copy_id = CopyId::placeholder();
+                fn resolve_copy_id(tree_value: Merge<Option<TreeValue>>) -> CopyId {
+                    let Some(copy_id_merge) = tree_value.to_copy_id_merge() else {
+                        return CopyId::placeholder();
+                    };
+                    copy_id_merge
+                        .resolve_trivial(SameChange::Accept)
+                        .cloned()
+                        .flatten()
+                        .unwrap_or_else(CopyId::placeholder)
+                }
+                let left_copy_id = resolve_copy_id(select_left(&path)?);
+                let right_copy_id = resolve_copy_id(select_right(&path)?);
+                let copy_id = if left_copy_id == right_copy_id {
+                    left_copy_id
+                } else {
+                    CopyId::placeholder()
+                };
```

Comment:
Shouldn't we always just take the right side? Or always the left side? I don't know how the builtin merge tool works, so take this with a grain of salt.